interfaces/builtin/device_mapper_support.go

```diff
@@ -0,0 +1,79 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2020 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package builtin
+
+// Allow access to already prepared device-mapper devices, such as LVM, raid,
+// etc. To control device-mapper devices add a plug for the
+// `device-mapper-control` interface. If you want to control LVM devices
+// add a plug for the `lvm-control` interface.
+const deviceMapperSupportSummary = `allows access to device-mapper devices`
+
+const deviceMapperSupportBaseDeclarationPlugs = `
+  device-mapper-support:
+    allow-installation: false
+    deny-auto-connection: true
+`
+
+const deviceMapperSupportBaseDeclarationSlots = `
+  device-mapper-support:
+    allow-installation:
+      slot-snap-type:
+        - core
+    deny-auto-connection: true
+`
+
+const deviceMapperSupportConnectedPlugAppArmor = `
+# Description: Allow access to device-mapper devices.
+
+@{PROC}/devices r,
+/run/udev/data/b[0-9]*:[0-9]* r,
+/sys/block/ r,
+/sys/devices/**/block/** r,
+
+# Access to LVM logical volume devices
+/dev/dm-[0-9]{,[0-9],[0-9][0-9]} rwk,                   # LVM LV (up to 1000 devices)
+
+# SCSI device commands, et al
+capability sys_rawio,
+
+# Perform various privileged block-device ioctl operations
+capability sys_admin,
+`
+
+var deviceMapperSupportConnectedPlugUDev = []string{
+	`SUBSYSTEM=="block"`,
+}
+
+type deviceMapperSupportInterface struct {
+	commonInterface
+}
+
+func init() {
+	registerIface(&deviceMapperSupportInterface{commonInterface{
+		name:                  "device-mapper-support",
```

Comment:
"support" is used for interfaces that allow to implement some functionality, so this name would be for something that implement the device-mapper outside the kernel? does that even make sense?

if the doc comment is correct this should be called "device-mapper-devices" instead?

Reply:
Have no issue with that, the reason for using the `-support` and `-control` names was after being involved in previous discussions around the `fuse-support` and (not yet implemented) `fuse-control` interfaces which from my pov give access to similar things. But when you point it out, this is more similar to the `block-devices` interface, so mimicking that naming makes more sense.

I'll put up a new patch set using the `device-mapper-devices` name.